### PR TITLE
Improved robustness for locating multiple result files

### PR DESCRIPTION
**Pull request by tskille**

### CODE REVIEW

File: opm/io/eclipse/ESmry.hpp [hunk 1]
```diff
@@ -137,6 +137,7 @@ class ESmry
     std::string read_string_from_disk(std::fstream& fileH, uint64_t size) const;
     void inspect_lodsmry();
     void Load_from_lodsmry(const std::vector<int>& keywIndVect) const;
+    bool is_number(const std::string& numstr) const;
```
Comment (joakim-hove) [inline opm/io/eclipse/ESmry.hpp:140]: This is not really an interesting method on the ESmry class - instead I would just implement the `is_number()` function in an anonymous namespace in the `.cpp` file.

Comment (tskille) [reply opm/io/eclipse/ESmry.hpp]: A agree that this is not an interesting function and I'm a bit surprised that that I didn't find a standard c++ function for this. ESmry.cpp doesn't have an anonymous namespace and I'm a bit  reluctant to create one. Hence I choose to move this function to the EclUtil.cpp file where we have simular functions like (fileExists,  isEOF, trimr ..) . This file is already included in ESmry.cpp.

**Comment by joakim-hove** [comment]
jenkins build this please

File: opm/io/eclipse/ESmry.hpp (hunk 1 above)
Comment (joakim-hove) [reply opm/io/eclipse/ESmry.hpp]: > I'm a bit surprised that that I didn't find a standard c++ function for this.

I guess the standard way would be:
```C++
try {
     std::stol( string_value );
     return true;
} catch (const std::invalid_argument& exc) {
     return false;
} 
```

**Pull request merged**